### ISSUE TITLE
It is extremly easy to accidently uninstall nix package manager

### ISSUE THREAD

**Issue by dnkmmr69420** [NixOS/nix#8124]
**Describe the bug**

If you run `nix-env -e '.*'` as root, you will uninstall nix itself. It is that easy to break nix. Same with nix profile remove too. You can also very easily uninstall nix. I think this should be considered a security flaw as well too.

**Steps To Reproduce**

1. as root type `nix-env -e '.*'` or `nix profile remove 0` or `nix profile remove 1` as root
2. Press enter
3. Now nix is completely destroyed

**Expected behavior**
Nix should stop itself from destroying instself

it should give an error that you are unable to uninstall nix. Nix should be uninstalled by typing `sudo rm -rf /nix` not by the nix package manager.

**Additional context**
I wanted to reset nix but I broke nix by typing `nix-env -e '.*'`

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by SuperSandro2000** [comment]
> I think this should be considered a security flaw as well too.

Why so?

**Comment by dnkmmr69420** [comment]
because it is so easy to accidently brick nix just by accidently typing 0 or 1 when removing a flake

**Comment by SuperSandro2000** [comment]
You should still be able to run nix from the nix store if that wasn't garbage collected yet.

Also it is generally pretty easy to brick things when you are root, but maybe something can be done here to prevent this.

**Comment by dnkmmr69420** [comment]
Exactly. there should be something to prevent that. You might be wanting to clean the root side of typing `nix-env -e '.*' but it deletes itself. the same should happen to nix profile where you can't easily break your system like that.
